### ISSUE TITLE
BigStitcher incompatible with other Fiji plugins due to dependency version skew

### ISSUE THREAD

**Issue by imagejan** [PreibischLab/BigStitcher#7]
Currently BigStitcher depends on (and is shipped by its update site together with) `SPIM_Registration-6.5.2.jar` which breaks plugins like _Descriptor-based registration_ that depend on `SPIM_Registration-5.0.16`.

This is confusing for users trying out BigStitcher. It should at least be mentioned in the [documentation on the wiki](https://imagej.net/BigStitcher).

Also, if you search http://maven.imagej.net for `SPIM_Registration`, the latest version seems to be `5.0.18-SNAPSHOT`, since all the newer development seems to happen [on a topic branch](https://github.com/fiji/SPIM_Registration/tree/accumulative-stitching) only. Are there any plans to change this any time soon?
Also, what are the plans of integrating BigStitcher into a regular Fiji? Is that something that will likely happen within the next months (e.g. at the December Dresden hackathon?), or should we rather keep suggesting people to use a separate Fiji installation if they want to try BigStitcher?

Thanks again for the awesome plugin development! 😄

**Comment by ctrueden** [comment]
@StephanPreibisch and I did some work recently to address this issue. IIRC, it should no longer be necessary to have the forked `6.5.2` version of `SPIM_Registration` anymore. At least, that dependency no longer exists on `BigStitcher` master. Did you try the latest SNAPSHOT, @imagejan?

**Comment by imagejan** [comment]
@ctrueden thanks for clarifying. I just tested with an up-to-date Fiji with the [`BigStitcher` update site](http://sites.imagej.net/BigStitcher/) enabled, which resulted in having `SPIM_Registration-6.5.2.jar` around.

I can test removing that version, I'll report back here...

**Comment by ctrueden** [comment]
@imagejan Yeah, I think the update site has not been updated in a while. The reason is mostly unrelated: since we switched BigStitcher to a SciJava command plugin (afc8223b79ec0aaa10f050749304a1d5810af41a), the BigStitcher menu appears way at the bottom of the Plugins menu, beneath the divider. So @StephanPreibisch was waiting for me to fix that problem. I will investigate it more deeply now.

**Comment by tpietzsch** [comment]
@ctrueden That's great! I would also rather have the BDV commands (that you made -- thank you 👍) stay in the same place in the menu where they are now.

**Comment by imagejan** [comment]
@ctrueden wrote:
>  the BigStitcher menu appears way at the bottom of the Plugins menu, beneath the divider.

As a workaround, you can define any (dummy) IJ1 menu entry using a `plugins.config` and place your jar file into `plugins`. This will ensure that the *BigStitcher* menu is created in the upper part of the *Plugins* menu, even all other menu entries are SciJava command plugins.

See https://github.com/imagej/imagej-legacy/issues/179#issuecomment-384964223.

@StephanPreibisch @hoerldavid Would this allow to go forward and bring the update site up to date with the latest release?

**Comment by StephanPreibisch** [comment]
Hi @ctrueden, we still didn't try this, is this still the suggest way to go? thanks a lot, Stephan

**Comment by ctrueden** [comment]
Sorry guys, but I have too much going on to worry about this issue right now. Will try to investigate in 2nd half of June. Or at the Learnathon if someone attending bugs me about it.

**Comment by ctrueden** [comment]
Has anyone had time to follow up on this issue yet? Is it still a problem?
